app/models/upload.rb

```diff
@@ -72,6 +72,18 @@ def self.last_uploads_rows
     uploads.sort_by { |upload| upload.csv_type.downcase }
   end
 
+  def self.latest_uploads
+    preview = Version.current_preview
+    production = Version.current_production
+    return [] if production.blank?
+
+    created_at = preview.present? && preview.number > production.number ? preview.created_at : production.created_at
+
+    Upload.last_uploads
```

Comment:
You don't have to convert this into an array first.  You can do it all with SQL:
`Upload.last_uploads.where("updated_at > ?", created_at)`

Reply:
changed